### ISSUE TITLE
Make healthy_panic_threshold configurable in config files

### ISSUE THREAD

**Issue by christian-posta** [envoyproxy/envoy#2463]
Right now, seems we can only configure the values from (https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_runtime#core) by runtime file-system files.  

To make envoy circuit breaking/outlier detection behave a little more like hystrix from netflix, we should be able to configure the healthy_panic_threshold a little easier. Setting the value to "0" causes envoy to behave like hystrix. If we can configure this a little easier (i.e., in config file) that would make life easier.

**Comment by christian-posta** [comment]
But.. before doing this.. I'd like @mattklein123 et.al opinion about the usefulness of the panic in production vs shutting off the calls completely when all endpoints have been ejected. Is this simply a "hey, things look really bad and I don't trust myself/my judgement here -- let's just continue to send traffic" kind of situation?

**Comment by mattklein123** [comment]
Personally, I would never run without panic enabled, at probably 50% which is the default. It's very easy to get into a cascading failure situation due to health check failure. With that said having a config option to set the % is very reasonable.

**Comment by christian-posta** [comment]
thx Matt. So I guess the difference with Hystrix is the expectation that when the circuit breaker is "opened" ie... any call to the upstream is immediately not made and 5xx returned.. and the app is supposed to "fallback" to some alternative path... 

With envoy that would look like the equivalent of zero healthy hosts in the LB pool and not panicking and the app responding to the 5xx with an alternative path in their code, etc.

**Comment by mattklein123** [comment]
> With envoy that would look like the equivalent of zero healthy hosts in the LB pool and not panicking and the app responding to the 5xx with an alternative path in their code, etc.

Yup exactly. Of course we could eventually implement some kind of retry policy which actually calls a different cluster, etc., but that would be a large amount of work.

**Comment by mattklein123** [comment]
I will fix this.

**Comment by mattklein123** [comment]
https://github.com/envoyproxy/data-plane-api/pull/465

**Comment by josdotso** [comment]
I'm having trouble with this with the latest (by hash not tag) docker hub image.

Is this wrong syntax?

```
    - name: example
      connect_timeout_ms: 300
      type: static
      lb_type: round_robin
      common_lb_config:
        healthy_panic_threshold:
          value: 49
      hosts:
        - url: tcp://127.0.0.1:41091
        - url: tcp://127.0.0.1:51091
```

Getting 
```
source/server/server.cc:71] error initializing configuration '/app/envoy.yaml': JSON at lines 0-0 does not conform to schema.
 Invalid schema: #
 Schema violation: additionalProperties
 Offending document key: #/common_lb_config
```

**Comment by gokulvanan** [comment]
@josdotso  In case you had not resolved it try:
`common_lb_config: { healthy_panic_threshold: { value: 35 } }`

This worked for me

**Comment by leecalcote** [comment]
Exposure of the panic threshold in config is great. Is the ability to define fallback paths for open circuits still a TBD capability?

**Comment by drao9** [comment]
Can this value be 0 ex: `common_lb_config: { healthy_panic_threshold: { value: 0 } }` in envoy 1.9.1? When I set it to 0. I get ```"common_lb_config": {
        "healthy_panic_threshold": {}
       }
      },``` in config_dump
